src/ui/AdvancedSearch/AdvancedSearch.ts

```diff
@@ -166,36 +167,67 @@ export class AdvancedSearch extends Component {
 
   private handlePopulateBreadcrumb(args: IPopulateBreadcrumbEventArgs) {
     if (this.needToPopulateBreadcrumb) {
-      const elem = $$('div', {
-        className: 'coveo-advanced-search-breadcrumb'
-      });
-      const title = $$('span', {
-        className: 'coveo-title'
-      });
-      title.text(l('FiltersInAdvancedSearch') + ' : ');
-      const clear = $$(
-        'span',
-        {
-          className: 'coveo-advanced-search-breadcrumb-clear'
-        },
-        SVGIcons.icons.checkboxHookExclusionMore
-      );
-      SVGDom.addClassToSVGInContainer(clear.el, 'coveo-advanced-search-breadcrumb-clear-svg');
-      clear.on('click', () => {
-        this.handleClearBreadcrumb();
-        this.usageAnalytics.logSearchEvent<IAnalyticsNoMeta>(analyticsActionCauseList.breadcrumbAdvancedSearch, {});
-        this.queryController.executeQuery();
-      });
+      const { container, title, clear } = this.buildBreadcrumbElements();
 
-      elem.append(title.el);
-      elem.append(clear.el);
+      container.append(title.el);
+      container.append(clear.el);
 
       args.breadcrumbs.push({
-        element: elem.el
+        element: container.el
       });
     }
   }
 
+  private buildBreadcrumbElements() {
+    return {
+      container: this.buildBreadcrumbContainer(),
+      title: this.buildBreadcrumbTitle(),
+      clear: this.buildBreacrumbClear()
+    };
+  }
+
+  private buildBreadcrumbContainer() {
+    return $$('div', {
+      className: 'coveo-advanced-search-breadcrumb'
+    });
+  }
+
+  private buildBreadcrumbTitle() {
+    return $$(
+      'span',
+      {
+        className: 'coveo-advanced-search-breadcrumb-title'
+      },
+      l('FiltersInAdvancedSearch') + ' : '
+    );
+  }
+
+  private buildBreacrumbClear() {
+    const clear = $$(
+      'span',
+      {
+        className: 'coveo-advanced-search-breadcrumb-clear'
+      },
+      SVGIcons.icons.checkboxHookExclusionMore
+    );
+
+    SVGDom.addClassToSVGInContainer(clear.el, 'coveo-advanced-search-breadcrumb-clear-svg');
+
+    const selectAction = () => {
+      this.handleClearBreadcrumb();
```

Comment:
I am not sure I get the exact meaning of "_Clear_" in _handleClearBreadcrumb()_ . Just for my understanding, would _handleUnselectedBreadcrumb_ be a good synonyme?